packages/cdk-s3-deployment/lib/lambda/handler.py

```diff
@@ -172,14 +183,39 @@ def cfn_send(event, context, responseStatus, responseData={}, physicalResourceId
     logger.info("| response body:\n" + body)
 
     headers = {
-        'content-type' : '',
-        'content-length' : str(len(body))
+        'content-type': '',
+        'content-length': str(len(body))
     }
 
     try:
-        request = Request(responseUrl, method='PUT', data=bytes(body.encode('utf-8')), headers=headers)
+        request = Request(responseUrl, method='PUT', data=bytes(
+            body.encode('utf-8')), headers=headers)
         with contextlib.closing(urlopen(request)) as response:
-          logger.info("| status code: " + response.reason)
+            logger.info("| status code: " + response.reason)
     except Exception as e:
         logger.error("| unable to send response to CloudFormation")
-        logger.exception(e)
\ No newline at end of file
+        logger.exception(e)
+
+def invalidation_failed(distribution_id, environment):
+    lambda_client = boto3.client('lambda')
+    lambda_payload = json.dumps({
+        "blocks": [
+            {
+                "type": "section",
+                "content": {
+                    "text": "CloudFront Invalidation for %s failed! (%s)" % (distribution_id, environment)
+                }
+            },
+            {
+                "type": "action",
+                "content": {
+                    "text": "AWS Console %s" % distribution_id,
+                    "url": "https://console.aws.amazon.com/cloudfront/home?region=us-east-1#distribution-settings:%s" % distribution_id
+                }
+            }
+        ],
+        "environment": "%s" % environment
+    })
+    lambda_client.invoke(FunctionName='serverless-slack-chatops-dev-sendSlack', 
```

Comment:
I'd rather we not invoke the function directly. I think this should publish to an SNS topic that way we can direct the message in multiple ways if we wanted. Also we don't want to hardcode function resource names.